lib/pengine/unpack.c

```diff
@@ -3727,10 +3735,20 @@ unpack_rsc_op(pe_resource_t *rsc, pe_node_t *node, xmlNode *xml_op,
         expired = true;
     }
 
+    old_rc = rc;
+    old_target_rc = target_rc;
+
     remap_operation(xml_op, rsc, node, data_set, on_fail, target_rc,
                     &rc, &status);
 
-    if (expired && (rc != target_rc)) {
+    maskable_probe_failure = !pe_rsc_is_bundled(rsc) && pcmk_xe_mask_probe_failure(xml_op);
+
+    if (expired && maskable_probe_failure && old_rc != old_target_rc) {
+        if (rsc->role <= RSC_ROLE_STOPPED) {
+            rsc->role = RSC_ROLE_UNKNOWN;
+            goto done;
+        }
```

Comment:
We should goto done here regardless of the rsc role

Reply:
Fixed and pushed.  I'm ready to see this merged.